### PR TITLE
Support no chromadb, sentence_transformers or pypdf

### LINKED ISSUE #531: ChromaDB required even if you use QdrantRetrieveUserProxyAgent and not Chroma
Considering that Chroma is unused by Qdrant, this shouldn't be required to install.

Error:
```zsh
ModuleNotFoundError                       Traceback (most recent call last)
File [~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/retrieve_user_proxy_agent.py:4](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/kenny/workspace/elapse/packages/research/notebooks/~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/retrieve_user_proxy_agent.py:4)
      [3](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/kenny/workspace/elapse/packages/research/notebooks/~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/retrieve_user_proxy_agent.py:3) try:
----> [4](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/kenny/workspace/elapse/packages/research/notebooks/~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/retrieve_user_proxy_agent.py:4)     import chromadb
      [5](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/kenny/workspace/elapse/packages/research/notebooks/~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/retrieve_user_proxy_agent.py:5) except ImportError:

ModuleNotFoundError: No module named 'chromadb'

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
/home/kenny/workspace/elapse/packages/research/notebooks/autogen-rag.ipynb Cell 4 line 1
----> [1](vscode-notebook-cell://wsl%2Bubuntu/home/kenny/workspace/elapse/packages/research/notebooks/autogen-rag.ipynb#W2sdnNjb2RlLXJlbW90ZQ%3D%3D?line=0) from autogen.agentchat.contrib.qdrant_retrieve_user_proxy_agent import QdrantRetrieveUserProxyAgent
      [3](vscode-notebook-cell://wsl%2Bubuntu/home/kenny/workspace/elapse/packages/research/notebooks/autogen-rag.ipynb#W2sdnNjb2RlLXJlbW90ZQ%3D%3D?line=2) ragproxyagent = QdrantRetrieveUserProxyAgent(
      [4](vscode-notebook-cell://wsl%2Bubuntu/home/kenny/workspace/elapse/packages/research/notebooks/autogen-rag.ipynb#W2sdnNjb2RlLXJlbW90ZQ%3D%3D?line=3)     name="ragproxyagent",
      [5](vscode-notebook-cell://wsl%2Bubuntu/home/kenny/workspace/elapse/packages/research/notebooks/autogen-rag.ipynb#W2sdnNjb2RlLXJlbW90ZQ%3D%3D?line=4)     human_input_mode="NEVER",
   (...)
     [14](vscode-notebook-cell://wsl%2Bubuntu/home/kenny/workspace/elapse/packages/research/notebooks/autogen-rag.ipynb#W2sdnNjb2RlLXJlbW90ZQ%3D%3D?line=13)     },
     [15](vscode-notebook-cell://wsl%2Bubuntu/home/kenny/workspace/elapse/packages/research/notebooks/autogen-rag.ipynb#W2sdnNjb2RlLXJlbW90ZQ%3D%3D?line=14) )

File [~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/qdrant_retrieve_user_proxy_agent.py:3](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/kenny/workspace/elapse/packages/research/notebooks/~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/qdrant_retrieve_user_proxy_agent.py:3)
      [1](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/kenny/workspace/elapse/packages/research/notebooks/~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/qdrant_retrieve_user_proxy_agent.py:1) from typing import Callable, Dict, List, Optional
----> [3](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/kenny/workspace/elapse/packages/research/notebooks/~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/qdrant_retrieve_user_proxy_agent.py:3) from autogen.agentchat.contrib.retrieve_user_proxy_agent import RetrieveUserProxyAgent
      [4](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/kenny/workspace/elapse/packages/research/notebooks/~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/qdrant_retrieve_user_proxy_agent.py:4) from autogen.retrieve_utils import get_files_from_dir, split_files_to_chunks
...
----> [6](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/kenny/workspace/elapse/packages/research/notebooks/~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/retrieve_user_proxy_agent.py:6)     raise ImportError("Please install dependencies first. `pip install pyautogen[retrievechat]`")
      [7](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/kenny/workspace/elapse/packages/research/notebooks/~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/retrieve_user_proxy_agent.py:7) from autogen.agentchat.agent import Agent
      [8](https://vscode-remote+wsl-002bubuntu.vscode-resource.vscode-cdn.net/home/kenny/workspace/elapse/packages/research/notebooks/~/.local/lib/python3.10/site-packages/autogen/agentchat/contrib/retrieve_user_proxy_agent.py:8) from autogen.agentchat import UserProxyAgent

ImportError: Please install dependencies first. `pip install pyautogen[retrievechat]
```

### PR DESCRIPTION
**Pull request by thinkall**
## Why are these changes needed?

Address #531 

Support no chromadb, sentence_transformers or pypdf when using customized vector database, embedding function or w/o pdf files, respectively.

## Related issue number

Closes #531

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by thinkall** [comment]
@kdcokenny could you check if this PR works for you? Thanks.

**Comment by 0xmatt1** [comment]
These things can be tricky, best to procede with caution

File: autogen/retrieve_utils.py [hunk 1]
```diff
@@ -240,6 +254,8 @@ def create_vector_db_from_dir(
     Returns:
         API: the chromadb client.
     """
+    if not HAS_CHROMADB:
+        raise ImportError("Please install dependencies first. `pip install pyautogen[retrievechat]`")
```
Comment (sonichi) [inline autogen/retrieve_utils.py:258]: make the error a constant in the header.

**Review by sonichi — COMMENTED** [review]
shall we decouple [retrievechat] and [chromadb]? Is there any common dependency between different retrieve backends? If not, we may create [retrievechat_chromadb] etc.
It will be good to clarify first, about what is required for all retrieve chat, what is required for each specific backend, etc.

**Comment by thinkall** [comment]
> shall we decouple [retrievechat] and [chromadb]? Is there any common dependency between different retrieve backends? If not, we may create [retrievechat_chromadb] etc. It will be good to clarify first, about what is required for all retrieve chat, what is required for each specific backend, etc.

RetrieveChat has no must-install dependencies. The [retrievechat] extras could be treated as a default solution for running it, but chromadb, sentence_transformers and pypdf all have alternatives, ipython is not needed for pure QA. Advanced users can install different dependencies as they want.

What if we keep the current [retrievechat] extras, but improve the docs  in the Installation section to guide users for different choices like qdrant for vectordb, huggingface or openai for embedding, unstructured for pdf/word... extraction?

**Comment by kdcokenny** [comment]
What I would suggest it adding import error try catch blocks which prompt the users to pip install chromadb or qdrant-client if importing one of them results in an import error. I can show an example if you’d like

**Comment by kdcokenny** [comment]
> @kdcokenny could you check if this PR works for you? Thanks.

When I was testing it I was getting weird errors like: `TypeError: Completions.create() got an unexpected keyword argument 'request_timeout'`

This may be due to the new openai v1 upgrade that I havent taken account for. Here's my exported py notebook:
```python
# %%
# %pip install -q pyautogen qdrant-client fastembed
# %pip install -q chromadb pypdf # required by autogen for some reason

# %%
%pip install -q git+https://github.com/microsoft/autogen.git@support_no_chromadb
%pip install -q qdrant-client fastembed

# %%
from qdrant_client import QdrantClient

host = "qdran...ws.com"

client = QdrantClient(host=host, api_key="...")

# %%
config_list = [
    {
        'model': 'gpt-3.5-turbo',
        'api_key': '...',
    },
]

# %%
# import autogen

# autogen.ChatCompletion.start_logging()

# %%
client.set_model("BAAI/bge-base-en-v1.5")

# %%
from autogen.agentchat.contrib.qdrant_retrieve_user_proxy_agent import QdrantRetrieveUserProxyAgent

ragproxyagent = QdrantRetrieveUserProxyAgent(
    name="ragproxyagent",
    human_input_mode="NEVER",
    max_consecutive_auto_reply=10,
    retrieve_config={
        "task": "qa",
        "chunk_token_size": 2000,
        "model": config_list[0]["model"],
        "client": client,
        "embedding_model": "BAAI/bge-base-en-v1.5",
        "collection_name": "wized_test",
    },
)

# %%
from autogen.agentchat.contrib.retrieve_assistant_agent import RetrieveAssistantAgent

assistant = RetrieveAssistantAgent(
    name="assistant", 
    system_message="You are a helpful assistant.",
    llm_config={
        "request_timeout": 600,
        "seed": 42,
        "config_list": config_list,
    },
)

# %%
assistant.reset()

qa_problem = "What is Wized?"
ragproxyagent.initiate_chat(assistant, problem=qa_problem)
```

**Pull request closed without merging**